solver/src/liquidity/uniswap.rs

```diff
@@ -125,25 +125,30 @@ impl UniswapLikeLiquidity {
 }
 
 impl Inner {
-    fn _settle(&self, input: (H160, U256), output: (H160, U256)) -> UniswapInteraction {
+    fn _settle(
+        &self,
+        (token_in, amount_in): (H160, U256),
+        (token_out, amount_out): (H160, U256),
+    ) -> UniswapInteraction {
+        let amount_in_with_slippage = slippage::amount_plus_max_slippage(amount_in);
```

Comment:
Question, do you think this amount should be capped at `sell_amount + fee_amount`?

Reply:
As long as the max_slippage is small enough (< 0.3%) this shouldn't be possible should it? Because manipulating the pool and moving it in one direction will force them to pay a 0.3% spread (unless they fully own the liquidity of the underlying Uniswap pool).

Is it easy enough to have access to the fee_amount here?

Reply:
I think its slippage + price movement, and not just slippage.

Reply:
Hmm, for some reason I thought that if `slippage + price_movement` was large enough, then a miner could extract value... I'm not 100% sure about this.